src/interfaces/IHooks.sol

```diff
@@ -54,14 +55,15 @@ interface IHooks {
     /// @param params The parameters for adding liquidity
     /// @param hookData Arbitrary data handed into the PoolManager by the liquidty provider to be be passed on to the hook
     /// @return bytes4 The function selector for the hook
-    /// @return BalanceDelta The hook's delta in token0 and token1. Positive: the hook is owed/took currency, negative: the hook owes/sent currency
+    /// @return HookReturnDelta The hook's delta in token0 and token1 as a BalanceDelta.
+    /// Positive: the hook is owed/took currency, negative: the hook owes/sent currency
     function afterAddLiquidity(
         address sender,
         PoolKey calldata key,
         IPoolManager.ModifyLiquidityParams calldata params,
         BalanceDelta delta,
         bytes calldata hookData
-    ) external returns (bytes4, BalanceDelta);
+    ) external returns (bytes4, HookReturnDelta);
```

Comment:
i think the liquidity return delta should stay as BalanceDelta

its quite weird that swap deltas are packed `(specified, unspecified)` while liquidity deltas are packed `(amount0, amount1)` and named differently than BalanceDelta

Reply:
if we decide to make liquidity-return-delta a BalanceDelta, we'll probably want to rename HookReturnDelta to HookSwapDelta?

Reply:
We're thinking about this differently but its an interesting point of view. I feel like as ever I'm thinking about this from the point of view of inside of core, and you're thinking about integrators which is perfect

What I wanted was to make this function easier to handle. Previously it would just generically work with an `int256` even though none of the hooks were actually working with an int256 which felt wrong. So I gave the name `HookReturnDelta` as a generic type for any delta returned by any of the custom accounting hooks. 

```
function callHookWithReturnDelta(IHooks self, bytes memory data, bool parseReturn)
        internal
-        returns (int256 delta)
+        returns (HookReturnDelta delta)
```

Then if you look inside the actual hook calls in `Hooks.sol` it then turns each `HookReturnDelta` into the type that is actually correct for that hook. Thats why the variable names for the function to create a `HookReturnDelta` were `int128 upperDelta, int128 lowerDelta` because those deltas could either be amount0 and amount1, OR specified and unspecified, depending whether the function in question needs a BalanceDelta or 2 int128s.

However I do see that while what I've created makes things simpler internally... it makes things more confusing for integrators. So I'm going to try to find a middle ground that works for everyone!